scripts/tag-and-version

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+const fetch = require("node-fetch");
+const exec = require("child-process-promise").exec;
+
+const packageJSON = require("../package.json");
+
+const Headers = Object.freeze({
+  Authorization: `token ${process.env.GITHUB_TOKEN}`,
+  "User-Agent": "HackerRank internal tools",
+});
+
+const Tags = Object.freeze({
+  Beta: "beta",
+  Stable: "stable",
+});
+
+const ExitCodes = Object.freeze({
+  Success: 0,
+});
+
+const BetaVersionRegex = /^[0-9]+\.[0-9]+\.[0-9]+\-[a-zA-z0-9.-]+$/;
+const StableVersionRegex = /^[0-9]+\.[0-9]+\.[0-9]+$/;
+
+/**
+ * Tests whether or not we are in `master` branch.
+ * @returns True if the current branch is `master`, else False
+ */
+function isMasterBranch() {
+  return process.env.BRANCH === "ref/heads/master";
+}
+
+/**
+ * Validates package version before release based on branch.
+ * @param isMaster - True if current branch is `master, else False.
+ * @returns True if version is valid against current branch, False otherwise.
+ */
+function validateVersion(isMaster) {
+  if (isMaster) {
+    return StableVersionRegex.test(packageJSON.version);
+  }
+
+  return BetaVersionRegex.test(packageJSON.version);
+}
+
+/**
+ * Provides with right release tag for package.
+ * @param isMaster - True if current branch is `master, else False.
+ * @returns Release Tag (stable/beta).
+ */
+function getReleaseTag(isMaster) {
+  return isMaster ? Tags.Stable : Tags.Beta;
+}
+
+/**
+ * Publish npm package with proper tag.
+ * @param tag - Tag to release the package (stable/beta).
+ * @returns A Promise that resolves once Exec is complete.
+ */
+async function publishToNPM(tag) {
+  const command = `npm publish --tag ${tag} --access public`;
```

Comment:
I'm not sure if this will propagate ENV variables correctly - [if you look here](https://github.com/interviewstreet/firepad-x/runs/2477036243?check_suite_focus=true#step:7:74) it looks like the `NODE_AUTH_TOKEN` is getting set for this command, so that publish 404s

Reply:
We can prepend the env var into command string, or is there much cleaner way?

Reply:
TBH I would like to have more of this logic in `yml` itself. And not to maintain this node script. What do you say?